rust/src/util.rs

```diff
@@ -134,7 +165,22 @@ impl Ord for Variant {
 
 impl Display for Variant {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        write!(f, "{}", self.name)?;
+        write!(f, "{}:{}", self.name, self.tag)?;
         Ok(())
     }
 }
+
+impl FromStr for Variant {
+    type Err = String;
```

Comment:
Strings shouldn't be used as errors due to many reasons (they do not implement `Error` trait, they are not convertible into super-error types, they can't be matched etc).

Reply:
Removed the entire `FromStr` implementation since it was not symmetrical to the `Display` implementation anymore